Adyen/Validators/BalanceChecker.swift

```diff
@@ -94,6 +102,8 @@ public struct BalanceChecker {
 }
 
 /// :nodoc:
-private func - (lhs: Payment.Amount, rhs: Payment.Amount) -> Payment.Amount {
-    Payment.Amount(value: lhs.value - rhs.value, currencyCode: lhs.currencyCode)
+internal func - (lhs: Payment.Amount?, rhs: Payment.Amount?) -> Payment.Amount {
```

Comment:
looks like we don't need a version with optional parameters, looks like it complies with the `lhs: Payment.Amount, rhs: Payment.Amount` signature.

Reply:
its needed for tests, I can make the optional version only in test target, but I don't see the benefit tbh

Reply:
i don't think tests should affect the code they are testing. this version of the function has unusual logic i think. it shouldn't be up to subtraction algorithm to unwrap optionals and provide default values.